lib/graphql/batch/loaded_field_extension.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+if Gem::Version.new(GraphQL::VERSION) < Gem::Version.new("1.9.0")
+  raise "GraphQL::Batch::LoadedFieldExtension is not supported on graphql gem versions less than 1.9"
+end
+
+require_relative 'loaded_field_extension/loader'
+
+module GraphQL::Batch
+  # Resolve the field using a class method on the GraphQL::Schema::Object
+  # for multiple instances. This avoids the need to extract the logic
+  # out into a {GraphQL::Batch::Loader} and automatically groups selections
+  # to load together.
+  #
+  # The class method must set the value on all given instances using an attribute
+  # writer of the same name as the resolver method.
+  #
+  # @example
+  #   class Product < GraphQL::Schema::Object
+  #     field :inventory_quantity, Int, null: false do
+  #       extension GraphQL::Batch::LoadedFieldExtension
+  #     end
+  #     def self.inventory_quantity(instances)
+  #       product_ids = instances.map { |instance| instance.object.id }
+  #       quantities = ProductVariant.group(:product_id).where(product_id: product_ids).sum(:inventory_quantity)
+  #       instances.each do |instance|
+  #         instance.inventory_quantity = quantities.fetch(instance.object.id, 0)
+  #       end
+  #     end
+  #
+  # For field selections to be loaded together, they must be given the same
+  # arguments. If the lookahead extra is used on the field, then it will group
```

Comment:
I'm not sure I follow this bit about the lookahead extra.

Reply:
For example, if we used a batch loaded field for `image` in the following GraphQL query,

```graphql
        {
          product1: product(id: "1") { image { id } }
          product2: product(id: "2") { image { filename } }
        }
```

then it might be fine to load the image field for both products if no lookahead or irep_node is used, since it won't depend on the selection set (`{ id }` or `{ filename }`).

However, if a lookahead were used (e.g. to avoid loading the image if only the `id` field is selected), then we don't want to load the image field for both products with a `lookahead` object that only represents `{ id }` and not actually load the `filename` for the second product.

Does that make sense?  If so, do you have any suggestions on how to clarify the documentation?